File: output/gemma4-e2b/avatar/Java/Python/codeforces_665_A.py
import sys

def solve():
    # Read input lines
    try:
        # Read a and tA
        line1 = sys.stdin.readline().split()
        if not line1:
            return
        a = int(line1[0])
        tA = int(line1[1])

        # Read b and tB
        line2 = sys.stdin.readline().split()
        if not line2:
            return
        b = int(line2[0])
        tB = int(line2[1])

        # Read timing
        timing = sys.stdin.readline().strip()

    except Exception:
        # Handle potential errors during reading
        return

    # Parse timing (HHMM)
    # Java: int hrs = Integer.parseInt ( timing.substring ( 0 , 2 ) ) ;
    # Java: int mins = Integer.parseInt ( timing.substring ( 3 ) ) ;
    try:
        hrs = int(timing[0:2])
        mins = int(timing[3:])
    except ValueError:
        # Handle case where timing format is incorrect
        return

    # Calculate simulation times
    simDeparture = hrs * 60 + mins
    simArrival = simDeparture + tA

    counter = 0
    
    # Loop: for ( int i = 300 ; i < 1440 ; i += b )
    # range(start, stop, step)
    for i in range(300, 1440, b):
        busDeparture = i
        busArrival = i + tB
        
        # if ( busDeparture >= simArrival || simDeparture >= busArrival ) continue ;

File: output/gemma4-e2b/avatar/Python/Java/codeforces_665_A.java
import java.util.Scanner;

public class codeforces_665_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read a and at
        String line1 = scanner.nextLine();
        String[] parts1 = line1.split(" ");
        int a = Integer.parseInt(parts1[0]);
        int at = Integer.parseInt(parts1[1]);

        // Read b and bt
        String line2 = scanner.nextLine();
        String[] parts2 = line2.split(" ");
        int b = Integer.parseInt(parts2[0]);
        int bt = Integer.parseInt(parts2[1]);

        // Read t1 and t2 (separated by ':')
        String line3 = scanner.nextLine();
        String[] parts3 = line3.split(":");
        int t1 = Integer.parseInt(parts3[0]);
        int t2 = Integer.parseInt(parts3[1]);

        // Calculations
        int st = t2 + (t1 - 5) * 60;
        int fin = st + at;

        int now = 0;
        int ans = 0;

        // Loop
        while (now < fin && now < 1140) {
            if (now + bt > st) {
                ans += 1;
            }
            now += b;
        }

        System.out.println(ans);
        
        scanner.close();
    }
}